velox/vector/tests/DecodedVectorTest.cpp

```diff
@@ -62,30 +62,41 @@ class DecodedVectorTest : public testing::Test, public VectorTestBase {
       const SelectivityVector& selection,
       SimpleVector<T>* outVector,
       bool dbgPrintVec) {
-    DecodedVector decoded(*outVector, selection);
-    auto end = selection.end();
-    ASSERT_EQ(expected.size(), end);
+    auto check = [&](auto& decoded) {
```

Comment:
This change suggests that 'selection' must always represent a continues set of rows from 0 to selection.end(). If that's the case, it might be cleaner to change the signature to take the number rows instead of SelectivityVector.

Reply:
yeh thats the case for now, let me change the test code.

Reply:
the test is written in a way that avoids reconstructing those each time, I add a check that selection.isAllSelected();